File: Collectibles.Frontend/collectibles/src/types/UserCollectible.ts
export interface UserCollectible {
    collectibleId: string;
    active?: boolean;
    description: string;
    collectedAt: string;
    imageUrl: string;
    bonusAchieved?: boolean;
}

export interface PutCollectibleRequest {
    active?: boolean;
    description: string;
    collectedAt: string;
    imageUrl: string;
    bonusAchieved?: boolean;
}

File: Collectibles.Backend/Collectibles.Backend.Api/Models/UserCollectible.cs
namespace Collectibles.Backend.Api.Models;

public class UserCollectible
{
    public Guid UserId { get; set; }
    public string CollectibleId { get; set; }
    public bool? Active { get; set; }
    public string Description { get; set; }
    public DateTime CollectedAt { get; set; }
    public string ImageUrl { get; set; }
    public bool? BonusAchieved { get; set; }
}

public class PutCollectibleRequest
{
    public bool? Active { get; set; }
    public string Description { get; set; }
    public DateTime CollectedAt { get; set; }
    public string ImageUrl { get; set; }
    public bool? BonusAchieved { get; set; }
    public string CollectionId { get; set; }
}